src/Microsoft.AspNet.Mvc.ViewFeatures/Buffer/MemoryPoolViewBufferScope.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc.ViewFeatures.Buffer
+{
+    /// <summary>
+    /// A <see cref="IViewBufferScope"/> that uses pooled memory.
+    /// </summary>
+    public class MemoryPoolViewBufferScope : IViewBufferScope, IDisposable
+    {
+        public static readonly int SegmentSize = 512;
```

Comment:
Yeah, we could go pretty low with this. I'd say let's check in this design change, and then experiment with tuning this.